### DISCUSSION TITLE
Dicom file with missing  sop_instance_uid

### DISCUSSION THREAD

**Discussion by sh4hri4roo** [Enet4/dicom-rs#252] [Q&A] [answered]
Hello,
I'm trying to work with Meta table and add "media_storage_sop_instance_uid" element to dicom object. Is there any sample on how to add missing elements and tags / write to existing meta table?

**Comment by Enet4** [comment, marked as answer]
Sorry, I must have overlooked this discussion. This has been answered in #255:

> The file is clearly missing mandatory fields as per [PS3.10 Chapter 7](https://dicom.nema.org/medical/dicom/current/output/chtml/part10/chapter_7.html), but it can be made more resilient with the following two changes:
> 
> 1. FileMetaTable expects mandatory fields to always be present, so it will have to be changed so that they also sit behind Options. Better coordinate this with https://github.com/Enet4/dicom-rs/issues/70 too.
> 2. File opening options can be extended to support strict/relaxed checking (whether to error on missing attributes); and to fill in missing file meta attributes with default values or overrides.
> 
> Until something like the above is implemented, DICOM-rs requires all Type 1 file meta information attributes to be present in a file, even if just placeholders.
